webapp/platform/client/src/client4.ts

```diff
@@ -4221,6 +4222,32 @@ export default class Client4 {
             {method: 'post', body: JSON.stringify(body)},
         )
     }
+
+    private validateClientURL(url?: string) {
+        const error = new ClientError(this.getUrl(), {
+            message: 'Invalid url.',
+            url,
+        })
+        if (!url) {
+            throw error;
+        }
+    
+        const path = url.split('?')[0].toLowerCase();
+        if (!path) {
+            throw error;
+        }
+    
+        if (path.includes('%2f')) {
+            throw error;
+        }
+    
+        const directories = path.split('/');
```

Comment:
Since this is the web, we'll also need to handle `\`. Browsers helpfully convert backslashes to forward slashes even though nobody really wants them to.